### DISCUSSION TITLE
Does vLLM return stream inference result in order?

### DISCUSSION THREAD

**Discussion by gaoxt1983** [vllm-project/vllm#3178] [General]
By in order, I mean, like this:
"I" -> "I like" -> "I Like vLLM" -> "I Like vLLM\n"

What I want to implement is that I want to stream inference returns only incremental characters, not whole output. So to achieve this goal, I must store the last output. I must ensure the last output is REAL last output, not out of ordered output.

Again, sorry of my poor English. :)

**Comment by simon-mo** [comment]
The streaming output is ordered and token-by-token. We follow the same semantics as OpenAI API
